### ISSUE TITLE
Hotkey Targeting UI Is bad and it should feel bad

### ISSUE THREAD

**Issue by z64555** [scp-fs2open/fs2open.github.com#2921]
Stems from investigation into PR #2911 

TLDR: The hotkey UI is a real mess and needs a rework to get it useful.

As reported by Phantom Hoover, a longstanding bug with the hotkey restore code effectively nukes mission default assignments on restarts, even if the player did not change the hotkey assignments

* There is a restart button, does it revert all changes to mission defaults reliably?
  * Question left unanswered, but the restart button should revert to mission defaults, regardless of restart or first-start status.

* Hotkey assignments should be treated like a delta applied on top of the mission defaults.  Mission defaults are loaded in first, player changes are applied on top of it when restore code functions.  Currently, the mission defaults are ignored entirely if there's any player changes.

* There is a 15 second limit from the start of the mission for players to be able to assign their own hotkey assignments and have it loaded again at restart.
  * Nowhere is this limit stated.  The player should be informed of this time limit.
  * The time limit is there, in theory, to prevent players from having an "unfair advantage" of assigning hotkeys to ships they shouldn't know about at the start of the mission (assuming one-time completion designs).

  * Mission designers may actually want the player to be able to assign hotkeys to ships that appear later in the mission.  There should be a flag in the mission file to allow this.  This would be useful for "groundhog day" type missions that require multiple restarts in order to complete.

**Comment by z64555** [comment]
This check should removed from hud_target_hotkey_add_remove() and applied in the callers.  This function should only be concerned about toggling the added/removed status of the hotkey.
```
hudtarget.cpp:797
    // don't add mission file added hotkey assignments if there are player added assignments
    // already in the list
    if ( (how_to_add == HOTKEY_MISSION_FILE_ADDED) && NOT_EMPTY(plist) ) {
        for ( hitem = GET_FIRST(plist); hitem != END_OF_LIST(plist); hitem = GET_NEXT(hitem) ) {
            if ( hitem->how_added == HOTKEY_USER_ADDED )
                return;
        }
    }
```
Further down, hudtarget.cpp:829 should have the following line added after it:
```
		hitem->how_added = how_to_add;
```
As part of the algorithm rewrite to treat player hotkey assignments as a delta on top of mission defaults.
